src/transformers/visitors/orchestrationInternalFunctionCallVisitor.ts

```diff
@@ -507,10 +530,51 @@ FunctionCall: {
         state.circuitImport.push('true');
         else
         state.circuitImport.push('false');
-        const newNode = buildNode('InternalFunctionCall', {
+      let newNode;
+        if(parent.nodeType === 'VariableDeclarationStatement') {
+          if(!functionReferncedNode.node.returnParameters.parameters[0]._newASTPointer.isSecret)  {
+          const decNode = buildNode('VariableDeclarationStatement')
+          decNode.declarations.push(functionReferncedNode.node.returnParameters.parameters[0]._newASTPointer);
+          decNode.interactsWithSecret = true;
+          decNode.declarations[0].declarationType = 'state';
+          decNode.declarations[0].isAccessed = true;
+          decNode.declarations[0].interactsWithSecret = true;
+          state.decNode ??= [];
+          const decNodeNames = state.decNode.map(node => node.declarations[0].name);
+          decNodeNames.includes(decNode.declarations[0].name) ? state.decNode : state.decNode.push(decNode);
+          }
+          const returnPara = functionReferncedNode.node.returnParameters.parameters[0].name;
+          let includeExpressionNode = false;
+          // this functions checks if the parent node interact with secret in the calling function or not
+          callingfnDefIndicators[parent.declarations[0].id].interactsWith.forEach( node => {
+            if(node.key != 'arguments' && node.interactsWithSecret)
+            includeExpressionNode = true;
+           })
+          functionReferncedNode.node.body.statements.forEach(exp => {
+            // If the return para interacts with public only in the internal function but with secret in calling function we need this expression in calling function
+          if(exp?.expression.leftHandSide?.name === returnPara && !exp.expression.leftHandSide.interactsWithSecret){
+            state.initNode = buildNode('BinaryOperation', {
```

Comment:
I think this will cause the same problem as state.initNode will be overwritten for each exp.

Reply:
yeah it doesn't matter, if it is overwritten as we will not use it outside this loop